### PR TITLE
feat(groups): adding support for groups form items

### PR DESCRIPTION
**Pull request by boc-tothefuture**

### CODE REVIEW

**Comment by boc-tothefuture** [comment]
Addresses #185  and #181 

To solve this it creates a def_item_delegator which delegates a predefined (and configurable) set of methods to an openhab Item.

File: lib/openhab/dsl/items/string_item.rb [hunk 1]
```diff
@@ -15,13 +15,13 @@ class StringItem
...
         include Comparable
-        include OpenHAB::DSL::Items::ItemDelegate
+        extend OpenHAB::DSL::Items::ItemDelegate
 
         # @return [Regex] Regular expression matching blank strings
         BLANK_RE = /\A[[:space:]]*\z/.freeze
         private_constant :BLANK_RE
 
-        def_delegator :@string_item, :to_s
+        def_item_delegator :@string_item, :to_s
```
Comment (jimtng) [inline lib/openhab/dsl/items/string_item.rb:24]: Why not use the default which includes groups?

Comment (boc-tothefuture) [reply lib/openhab/dsl/items/string_item.rb:24]: Because I missed that one.  :)

**Review by pacive — APPROVED** [review]
Looks good!

**Comment by boc-tothefuture** [comment]
Made the one change you spotted @jimtng  - this look good to merge?

**Review by jimtng — APPROVED** [review]
LGTM

**Pull request merged**